infra/gravity/provision.go

```diff
@@ -353,11 +354,34 @@ func bootstrapAzure(ctx context.Context, g *gravity, param cloudDynamicParams) (
 
 // bootstrapCloud is a simple workflow to wait for cloud-init to complete
 func bootstrapCloud(ctx context.Context, g *gravity, param cloudDynamicParams) (err error) {
-	err = sshutil.WaitForFile(ctx, g.Client(), g.Logger(), cloudInitCompleteFile, sshutil.TestRegularFile)
-	if err != nil {
-		return trace.Wrap(err)
-	}
+	return trace.Wrap(g.waitForBootstrapScript(ctx))
+}
 
+func (g *gravity) waitForBootstrapScript(ctx context.Context) error {
+	cmd := fmt.Sprintf(`sudo bash -c "[[ -f %v ]] && exit 0 || [[ -f %v ]] && exit 2 || exit 1"`,
+		cloudInitCompleteFile, cloudInitFailedFile)
```

Comment:
In order to fail safe, we want `/var/lib/bootstrap_failed` to take precedence over `/var/lib/bootstrap_complete` if both are present.  This should never happen, but it is safer to fail if it does.

On a meta note, it feels "heavy" to put  business logic in inline bash (similar to your [earlier remark](https://github.com/gravitational/robotest/pull/206#discussion_r409442240) about go templating in make/bash).  In most of the rest of robotest we shell out to run a command, and then interpret the results in go.  Here we have mild flow control.

Because it is so mild, I'm happy to take this bit as is. In general I want to avoid creating more chimeras like `file_transfer.go`

Reply:
Yes, I understand and agree and yet I wanted to batch the requests and did not have a better idea rather than come up with another bash script.
Cleaner would probably be to have a direct way to ask the system about the status of the startup script. Yet another way would be using cloud-init so instances are created properly initialized (although I'm not super certain this is the cleanest solution either).
I still ponder how robotest nodes would [blacklist](https://github.com/gravitational/robotest/pull/138) the runner on the ground of its erratic connect behavior.

Reply:
Fixed in 605b429.